### PR TITLE
Add Api Management Basic Sku

### PR DESCRIPTION
**Pull request by solankisamir**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by KedarJoshi** [comment]
#sign-off

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimdeployment.json [hunk 1]
```diff
@@ -981,7 +981,8 @@
           "enum": [
             "Developer",
             "Standard",
-            "Premium"
+            "Premium",
+            "Basic"
```
Comment (fearthecowboy) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimdeployment.json]: Has SDKs for this been published already?

Comment (solankisamir) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimdeployment.json]: not yet.

Comment (fearthecowboy) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimdeployment.json]: oh, good.

**Comment by solankisamir** [comment]
@fearthecowboy ping

**Comment by fearthecowboy** [comment]
@solankisamir  Sorry;  I restarted the validation and then forgot to come back.

The validator is complaining about a couple properties that aren't camel-cased.
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/311558732#L702

Can we clean those up while we're here?

(the breaking-change detection is complaining that there are changes without an  API change, but you've assured me this hasn't had an SDK published yet)

**Comment by solankisamir** [comment]
Issue with Value contract will be addressed in next version 2018-01-01
https://github.com/Azure/azure-rest-api-specs/issues/1987

Issue with certificate_password will be addressed in next version 2018-01-01
https://github.com/Azure/azure-rest-api-specs/issues/1881

We have active issues tracking that.

**Review by fearthecowboy — APPROVED** [review]

**Pull request merged**